### DISCUSSION TITLE
[面試實戰D1] 為何用 local storage 裡的東西做驗證，那就不會有 CSRF 的問題?

### DISCUSSION THREAD

**Discussion by BoisonChang** [Lidemy/forum#34] [General]
上網查不太到答案，甚至很多說反而會被盜用。

> 原因是因為假如我們的前端網頁如果很容易存取到這些資訊的話，相對的它也容易暴露在風險之中！例如 localStorage 和 sessionStorage 雖然容易在瀏覽器內使用 js 操作，但假如網站被植入惡意 js 程式碼，你放在 localStorage 上的東西就會被盜用了（這個手法叫做 XSS 攻擊）。因此像是一些 credential 的東西，其實是不適合放在 localStorage 和 sessionStorage 內。

還沒上到 22 週，可以簡單跟我解釋或者跟我說 Lidemy 哪 Part 哪一節有提到這塊嗎？

感恩><

**Comment by aszx87410** [comment]
你要想一下 CSRF 之所以可以成立的原因是什麼，是因為你可以讓 server 誤以為是真的使用者在發送資訊，而這個前提就是，攻擊者從 A 網站往 B 網站發 request 跟使用者自己從 B 網站發 request 到 server，所攜帶的資訊是一樣的。

例如說 cookie，因為這兩者都有 cookie 所以伺服器辨別不出來，但 cookie 是瀏覽器幫忙帶上的，所以只要瀏覽器不幫忙帶（例如說 SameSite）攻擊就失效了

同理，localStorage 不是瀏覽器會自動幫忙帶的東西，所以就天然免疫 CSRF
